cmd/bundlectl/version.bzl

```diff
@@ -0,0 +1 @@
+VERSION = "1"       # What should be the real source of truth for the version?  Project file, git commit, etc? 
```

Comment:
I don't know bazel that well. Is there a way to:

1) add in the git-commit, probably as a separate field
2) if the build is done in a dev env with uncommitted changes, then it should be flagged as dirty; e.g. if it is commit ab123c but changes are there, then this field would read "ab123c-dirty".

Reply:
Not at my desk, but looks like there's some variables bazel references.
I'll merge it now and add that in after when I get back. 

I'm not sure we'll be able to do the dirty flag thing since I'm unsure how bazel would know if it's committed to git vs the remote server, and having it marked dirty because it's not committed locally is pointless.

Reply:
I think I wasn't super clear before, which is my bad. I would rather we don't use bazel for the mechanics of `bundlectl version`. This repository is pretty bazel-lite, I would almost consider getting rid of bazel except that it's useful for things like prow-infra, license gathering, etc.

The disadvantage of relying on bazel, is that it makes `go install` not work. You have to build the binary with bazel. I think that would be a little sad, and also would be a bit more burden on us, because it would probably mean we would need to pre-build a bunch of binaries of different architectures.

By the way, here's what kubectl does:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/version/version.go
-> 
https://github.com/kubernetes/kubernetes/tree/master/pkg/version
-> 
https://github.com/kubernetes/kubernetes/blob/master/hack/lib/version.sh

I think we could do something similar with go-codegen, makefile, or something like hack/update-version.sh

Reply:
The bazel flag is just the equivalent to the ldflag build option which we can pass.   It's easy to solve.  No big deal.

The only question that might be interesting is go get with the git commit hash, due to the fact that we'd have to make go get somehow able to see the git commit hash, which I'm not sure go get is actually aware of that? and at best we could in source check in one hash off, which doesn't give guarantees about how much that hash differs.